src/omap.rs

```diff
@@ -411,10 +412,10 @@ impl<'s> AddressMap<'s> {
     /// This iterator traverses all mapped ranges in the order of the PDB-internal mapping. All
     /// empty or eliminated ranges are skipped. Thus, the iterator might be empty even for non-empty
     /// ranges.
-    pub fn rva_ranges(&self, start: PdbInternalRva, end: PdbInternalRva) -> RvaRangeIter<'_> {
+    pub fn rva_ranges(&self, range: Range<PdbInternalRva>) -> RvaRangeIter<'_> {
```

Comment:
Index doesn't work because it has to return a reference to some output type. 

Also, I think the `Index` trait generally carries the notion of a simple lookup or slice operation, but what we do is more of a dynamic mapping which - at least in my mind 😄 - doesn't really fit the semantics of the trait.